llpc/util/llpcCacheAccessor.cpp

```diff
@@ -0,0 +1,102 @@
+/*
+ ***********************************************************************************************************************
+ *
+ *  Copyright (c) 2021 Google LLC All Rights Reserved.
+ *
+ *  Permission is hereby granted, free of charge, to any person obtaining a copy
+ *  of this software and associated documentation files (the "Software"), to deal
+ *  in the Software without restriction, including without limitation the rights
+ *  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ *  copies of the Software, and to permit persons to whom the Software is
+ *  furnished to do so, subject to the following conditions:
+ *
+ *  The above copyright notice and this permission notice shall be included in all
+ *  copies or substantial portions of the Software.
+ *
+ *  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ *  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ *  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ *  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ *  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ *  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ *  SOFTWARE.
+ *
+ **********************************************************************************************************************/
+/**
+ ***********************************************************************************************************************
+ * @file  llpcCacheAccessor.cpp
+ * @brief LLPC source file: Implementation of a class that will create an interface to easily check the caches that need
+ *to be checked. (independent of LLVM use).
+ ***********************************************************************************************************************
+ */
+#include "llpcCacheAccessor.h"
+#include "llpcCompiler.h"
+#include "llpcContext.h"
+
+#define DEBUG_TYPE "llpc-cache"
+
+namespace Llpc {
+
+// =====================================================================================================================
+// Access the given caches using the hash.
+//
+// @param context : The context that will give the caches from the application.
+// @param hash : The hash for the entry to access.
+// @param compiler : The compiler object with the internal caches.
+CacheAccessor::CacheAccessor(Context *context, MetroHash::Hash &cacheHash, Compiler *compiler) {
```

Comment:
Can `context` or `compiler` be null? `context` it immediately accessed, while `compiler` is dereferenced by `initialize`, so could we add assertions or turn these into references?